### PR TITLE
move test data to zenodo

### PR DESCRIPTION
**Pull request by atravitz**
resolves https://github.com/OpenFreeEnergy/openfe/issues/1132

also resolves https://github.com/OpenFreeEnergy/openfe/issues/1130

Checklist
* [x] switch from using local data to fetching from zenodo
* [x] support running tests offline if data already exists in the cache
*  match existing `openfe fetch rbfe-tutorial-results` behavior: this will require a bit more work on the cli fetching side, I'll address in a follow-up PR

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Comment by mikemhenry** [comment]
Lets add an import of `from .conftest import HAS_INTERNET` and then a mark `@pytest.mark.skipif(not HAS_INTERNET, reason="Internet seems to be unavailable")` So that we don't get a test fail if someone runs this on an air gapped network

**Comment by atravitz** [comment]
my motivation for putting pooch in a fixture is that it won't get called for tests that don't require it. e.g. if you want to just run the `test_format_estimate_uncertainty` test, you dont want pooch downloading a test dataset.

**Comment by atravitz** [comment]
> Lets add an import of `from .conftest import HAS_INTERNET` and then a mark `@pytest.mark.skipif(not HAS_INTERNET, reason="Internet seems to be unavailable")` So that we don't get a test fail if someone runs this on an air gapped network

I'd like to support these tests in instances where the data exists in the cache, but `HAS_INTERNET=False`? This would cover most of us doing active development, so that we can 1) do offline development if we already have the data cached 2) test on air-gapped networks by manually uploading data into the cache. 

I can brute-force this, but I'm curious if you have an elegant way of going about this?

**Comment by mikemhenry** [comment]
Let me think about it -- my first thought is that we check if `HAS_INTERNET` or `OFE_USE_DATA_CACHE` (which we make a new envar for) is true, since if either of those are true then we can run the tests that need pooch, but the tests that check if we can fetch tutorial data don't use pooch so we would want to keep the way `HAS_INTERNET` behaves for them.

Can you test this?
1. Run a test that uses pooch
2. Turn off your internet
3. Run test again

And see if that actually works?

**Comment by atravitz** [comment]
> Let me think about it -- my first thought is that we check if `HAS_INTERNET` or `OFE_USE_DATA_CACHE` (which we make a new envar for) is true, since if either of those are true then we can run the tests that need pooch, but the tests that check if we can fetch tutorial data don't use pooch so we would want to keep the way `HAS_INTERNET` behaves for them.
> 
> Can you test this?
> 
>     1. Run a test that uses pooch
> 
>     2. Turn off your internet
> 
>     3. Run test again
> 
> 
> And see if that actually works?

Yes that works as expected!

File: openfecli/tests/commands/test_gather.py [hunk 1]
```diff
@@ -134,30 +135,29 @@ def test_get_column(val, col):
...
+@pytest.fixture
+def rbfe_result_dir()->pathlib.Path:
+    def _rbfe_result_dir(dataset)->str:
+        ZENODO_RBFE_DATA.fetch(f'{dataset}.tar.gz', processor=pooch.Untar())
+        cache_dir = pathlib.Path(pooch.os_cache('openfe'))/f'{dataset}.tar.gz.untar/{dataset}/'
+        return  cache_dir
+
+    return _rbfe_result_dir
+
+@pytest.mark.skipif(not os.path.exists(POOCH_CACHE) and not HAS_INTERNET,reason="Internet seems to be unavailable and test data is not cached locally.")
```
Comment (atravitz) [inline openfecli/tests/commands/test_gather.py:156]: @mikemhenry this works but misses edge cases where the openfe cache exists, but the test data inside has been deleted (I think this would only happen if a user manually messed with it).

Comment (mikemhenry) [reply openfecli/tests/commands/test_gather.py:156]: I don't think it really misses the edge case, since if someone did that then the test will fail since the data is missing

File: openfecli/tests/data/rbfe_results.tar.gz
Comment (atravitz) [inline openfecli/tests/data/rbfe_results.tar.gz:1]: see here: https://github.com/OpenFreeEnergy/openfe/pull/1121/files#r1964427513

same data, just correctly(?) compressed.

**Review by mikemhenry — APPROVED** [review]
LGTM -- I don't think we need a news entry since this is pretty back end stuff and functionally it should all work. The only thing I can think of is if we don't add it to the changelog, somewhere we need to document how to manually load the test data on an air gapped system.

**Comment by atravitz** [comment]
> LGTM -- I don't think we need a news entry since this is pretty back end stuff and functionally it should all work. The only thing I can think of is if we don't add it to the changelog, somewhere we need to document how to manually load the test data on an air gapped system.

https://github.com/OpenFreeEnergy/openfe/pull/1156 working on it here
